feat: slack background agent docs (#1463)
knocklabs/docs: content/integrations/extensions/slack.mdx

---
title: Connecting Knock to your Slack workspace
description: Use the Knock Slack extension to launch Knock agents from your Slack workspace.
layout: integrations
tags: ["slack", "extensions"]
section: Integrations > Extensions
---

> The Knock Slack extension lets your team ask questions about your Knock resources directly in Slack. Users can DM the agent or `@Knock` in any channel to kick off an agent session.

Reviewer: ```suggestion
The Knock Slack extension lets your team launch Knock agents directly in Slack to build, manage, and ship customer messaging. Users can DM the agent or tag `@Knock` in any channel to kick off an agent session.
```

Reviewer: Feel like we don't want to only lead with the 'questions' value prop. 'Build, manage, and ship customer messaging' is how we're positioning in changelog

## Setting up the Knock Slack extension

1. Go to **[Settings > Extensions](https://dashboard.knock.app/~/settings/integrations/extensions)** in the Knock dashboard.
2. Click "Connect" next to the Slack extension or go to the [installation page](https://control.knock.app/slack/install) here.
3. You'll be prompted to install the Knock app for Slack in your workspace.

> 4. After installing, you'll be redirect to Knock to finalize the installation.

Reviewer: ```suggestion
4. After installing, you'll be redirected to Knock to finalize the installation.
```

> 5. Start using the Slack extension by prompting the Knock agent in a direct message or `@Knock` in any channel.

Reviewer: ```suggestion
5. Prompt the Knock agent in a direct message or tag `@Knock` in any channel.
```

## Authenticating users

> In order to use the Knock Slack extension, users will need to link their Slack user to their Knock account.

Reviewer: ```suggestion
In order to use the Knock Slack extension, users will need to link their Slack user account to their Knock account.
```

> Users within your Slack workspace can authenticate by prompting the Knock agent in a direct message or `@Knock` in any channel. They will receive a DM with a button to authenticate. Clicking the button will take them to the Knock dashboard where they can authenticate their account.

Reviewer: ```suggestion
Users within your Slack workspace can authenticate by prompting the Knock agent in a direct message or tagging `@Knock` in any channel. They will receive a DM with a button to authenticate. Clicking the button will take them to the Knock dashboard where they can authenticate their account.
```

> Once linked, users can then disconnect their linked Slack account from their user settings under "Settings > Profile > Overview". Once disconnected, they will no longer be able to use the Knock Slack extension.

Reviewer: This is just the instructions to undo authentication, right? Wonder if we can stick this at the end, since it almost reads as another final step. Plus we want to inherently discourage people disconnecting the link

Reviewer: For example, new H2 section at bottom "Disconnecting Slack" with similar sentence: "Users can disconnect their linked Slack account from their user settings under "Settings > Profile > Overview" at any time. Once disconnected, they will no longer be able to use the Knock Slack extension."

## Using the Knock Slack extension

### Kicking off an agent

> Users can kick off an agent session by prompting the Knock agent in a direct message or `@Knock` in any channel. The agent will respond with a message indicating that it has started and will begin processing the user's request.

Reviewer: ```suggestion
Users can kick off an agent session by prompting the Knock agent in a direct message or tagging `@Knock` in any channel. The agent will respond with a message indicating that it has started and will begin processing the user's request.
```

> The Knock app will only be available in the public channels that it has been invited to. You can invite the Knock app to a channel by at-mentioning it in the channel, or using the "Add to channel" button in Slack.

Reviewer: Is it worth having an H3 section for "Adding Knock to channels"? And place it above 'Kicking off an agent"? Right now, someone might get stuck immediately if the Knock agent isn't in the channel yet.

Here are some example prompts you can use to kick off an agent session:

> - `@Knock create a new onboarding workflow that sends a welcome compaign to new users`

Reviewer: ```suggestion
- `@Knock create a new onboarding workflow that sends a welcome campaign to new users`
- `@Knock draft our May newsletter using the most recent changelogs from our website`
- `@Knock build a new in-app guide promoting a customer discount for our upcoming event`
```

Reviewer: Typo, plus a few more relevant examples

- `@Knock create a new audience of paid subscribers`
- `@Knock find all of my workflows that are sending emails`

### Modified resources

> If the agent session modifies one or more resources, you will see a link to the resource in the conversation history. Clicking the link will open the resource in the Knock dashboard.

Reviewer: ```suggestion
For every resource that the agent modifies, it will return a link to each resource, visible in the conversation history. Clicking a link will open the resource in the Knock dashboard.
```

### Specifying an environment

By default, the Knock Slack extension will operate in the `development` environment. You can specify a different environment by referencing it within your prompt. The Knock agent will attempt to automatically determine the environment based on the context of the prompt.

As an example, you can specify the environment in a prompt like this:

> > @Knock in the production environment, update the subject line of the welcome email to 'Welcome to our platform!'

Reviewer: ```suggestion
`@Knock in the production environment, update the subject line of the welcome email to 'Welcome to our platform!'`
```

### Viewing an agent session

> From a kicked off agent session, you can view the conversation history by clicking the "Open agent" button. This will open a new tab with the conversation history in the Knock dashboard. Any user within your Slack workspace can view the conversation history of an agent session by clicking the "Open agent" button.

Reviewer: ```suggestion
In each agent session, you can click the "Open agent" button to view the full conversation history in the Knock dashboard. Any user within your Slack workspace can click the "Open agent" button to view that thread's conversation history.
```
